src/main/credentialServer/httpCredentialServer.ts

```diff
@@ -0,0 +1,210 @@
+      const url = new URL(request.url as string, `http://${ request.headers.host }`);
+      const path = url.pathname;
+      const pathParts = path.split('/');
+      const [data, error] = await serverHelper.getRequestBody(request, MAX_REQUEST_BODY_LENGTH);
+
+      if (error) {
+        console.debug(`${ path }: write back status 400, error: ${ error }`);
+        response.writeHead(400, { 'Content-Type': 'text/plain' });
+        response.write(error);
+
+        return;
+      }
+      console.debug(`Processing request ${ method } ${ path }`);
+      if (pathParts.shift()) {
+        response.writeHead(400, { 'Content-Type': 'text/plain' });
+        response.write(`Unexpected data before first / in URL ${ path }`);
+      }
+      const command = this.lookupCommand(method, pathParts[0]);
+
+      if (!command) {
+        console.log(`404: No handler for URL ${ method } ${ path }.`);
+        response.writeHead(404, { 'Content-Type': 'text/plain' });
+        response.write(`Unknown command: ${ method } ${ path }`);
+
+        return;
+      }
+      await command.call(this, helperName, data, request, response);
+    } catch (err) {
+      console.log(`Error handling ${ request.url }`, err);
+      response.writeHead(500, { 'Content-Type': 'text/plain' });
+      response.write('Error processing request.');
+    } finally {
+      response.end();
+    }
+  }
+
+  protected lookupCommand(method: string, commandName: string): dispatchFunctionType|undefined {
+    if (commandName) {
+      return this.dispatchTable[method]?.[commandName];
+    }
+
+    return undefined;
+  }
+
+  async get(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireJSONOutput, helperName, 'get', data, request, response);
+  }
+
+  async list(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireJSONOutput, helperName, 'list', data, request, response);
+  }
+
+  async erase(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'erase', data, request, response);
+  }
+
+  async store(helperName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await this.doNamedCommand(requireNoOutput, helperName, 'store', data, request, response);
+  }
+
+  protected combineStrings(stdout: string, stderr: string): string {
+    if (!stdout) {
+      return stderr;
+    } else if (!stderr) {
+      return stdout;
+    } else {
+      return `${ stdout }\n\n${ stderr }}`;
+    }
+  }
+
+  protected async doNamedCommand(outputChecker: checkFunctionOutputType, helperName: string, commandName: string, data: string, request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
```

Comment:
I didn't wrap it because the indent style for multi-line function definitions is to add one indent at the start of the line; I prefer indenting under the open-paren (shifted one space to the right), but easy to do.